process/block/metablock.go

```diff
@@ -1628,6 +1633,33 @@ func (mp *metaProcessor) ApplyBodyToHeader(hdr data.HeaderHandler, bodyHandler d
 	return nil
 }
 
+func (mp *metaProcessor) verifyEpochStartDataForMetablock(metaBlock *block.MetaBlock) error {
+	if !metaBlock.IsStartOfEpochBlock() {
+		return nil
+	}
+
+	epochStart, err := mp.createEpochStartForMetablock(metaBlock)
```

Comment:
metablock is not used in method createEpochStartForMetablock, should this not take into consideration metaBlock because you are comparing the receivedMetaBlock "metaBlock" with what you are creating?

Reply:
done.